### ISSUE TITLE
Question about Mu/Log Variance to Latent code

### ISSUE THREAD

**Issue by mtkmark** [2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN#10]
The following section contains the math of latent code generation.
Where can we find " what exact math operation does it perform  " ? 

Thanks
====
    def reparameterize(self, mu, logvar):
        if self.training:
          std = logvar.mul(0.5).exp_()
          eps = Variable(std.data.new(std.size()).normal_())
===

**Comment by hui-po-wang** [comment]
Hi @mtkmark,

I am not sure if I have your question right, but let me make a guess....

According to two pictures below, the first figure shows that we should draw some noise from standard normal, and multiply it  by ```standard deviation``` and add ```mean``` to it, and in the sample codes, they indeed return values based on the operation. This is so-called reparameterization.

To achieve the goal, we first compute the standard deviation (For the reason why we can compute it by ``` logvar.mul(0.5).exp_() ```, please refer to the question I answered before https://github.com/2017-fall-DL-training-program/VAE-GAN-and-VAE-GAN/issues/9). As we get the standard deviation, we should draw some noise from normal distribution of which dimension is equal to the dim. of the standard deviation (also equal to the dim. of mean. It's straightforward because each normal distribution has one var and one mean as its parameters. In our case, it should be 10.) In the end, the function shown in the second figure returns the ```reparameterized noise```.

![screenshot from 2017-11-28 10-30-02](https://user-images.githubusercontent.com/10228478/33299567-2d6eef80-d427-11e7-913a-75303ffe7e74.png)
![screenshot from 2017-11-28 10-33-45](https://user-images.githubusercontent.com/10228478/33299674-a57f8d04-d427-11e7-932d-b500ba686499.png)
